File: archive/flutter-app/lib/core/ports/i_storage_port.dart
/// Outbound port: generic key/value persistence for user settings and
/// favorites. Web adapter backs this with LocalStorage/IndexedDB, desktop
/// adapter backs this with the native file system.
abstract interface class IStoragePort {
  Future<String?> read(String key);

  Future<void> write(String key, String value);

  Future<void> delete(String key);
}

File: app/src/core/ports/IStoragePort.ts
/**
 * Outbound port for persisting settings, prefs, and saved templates.
 * Mirrors `IStoragePort` in the Flutter reference app
 * (`lib/core/ports/i_storage_port.dart`). Implementations (Phase 6):
 * web → localStorage/IndexedDB, desktop → Tauri `fs` plugin.
 */
export interface IStoragePort {
  get(key: string): Promise<string | null>
  set(key: string, value: string): Promise<void>
  remove(key: string): Promise<void>
  keys(): Promise<string[]>
}
